pkg/codegen/go/gen_program.go

```diff
@@ -816,9 +816,6 @@ func (g *generator) genLocalVariable(w io.Writer, v *pcl.LocalVariable) {
 			assignment = "="
 		}
 	}
-	if name == "_" {
-		assignment = "="
-	}
```

Comment:
I assume this will introduce the opposite case where `_, err :=` is generated even when `err` is already declared, but that's ok because [short variable declarations may redeclare variables](https://go.dev/ref/spec#Short_variable_declarations)?

Reply:
I don't think so, we should handle that [here](https://github.com/pulumi/pulumi/blob/e3a60af8cb7a20ea68095d5fc77eb6c50452486a/pkg/codegen/go/gen_program.go#L831) still